src/scales/scale.linearbase.js

```diff
@@ -23,16 +25,26 @@ function generateTicks(generationOptions, dataRange) {
 	var precision = generationOptions.precision;
 	var spacing, factor, niceMin, niceMax, numSpaces;
 
+	var rmin = dataRange.min;
+	var rmax = dataRange.max;
+	var isNullOrUndef = helpers.isNullOrUndef;
+
 	// spacing is set to a nice number of the dataRange divided by maxNumSpaces.
 	// stepSize is used as a minimum unit if it is specified.
-	spacing = helpers.niceNum((dataRange.max - dataRange.min) / maxNumSpaces / unit) * unit;
-	numSpaces = Math.ceil(dataRange.max / spacing) - Math.floor(dataRange.min / spacing);
+	spacing = helpers.niceNum((rmax - rmin) / maxNumSpaces / unit) * unit;
```

Comment:
Could make this `var spacing` and remove that declaration above

Reply:
+1 and I would move the uninitialized ones (line 26) right after `var spacing = ...`. We could also drop the comments above, outdated for `stepSize` and useless for `spacing`.